cog/daily_charge.py

```
@@ -23,95 +23,257 @@ def get_channels():  # 取得特殊用途頻道的清單，這裡會用來判斷
+            self.bot,
+            charge_channel_id,
+            after=earliest_downtime_start,
+            before=latest_downtime_end,
+        )
+
+        # TODO: cache the last_charged date
+        # cached_change: UserDict = UserDict()
+        with mysql_connection() as c:
+            connection, cursor = c
+            # XXX: check with a better method, because the module on the running machine have no `_connection` attribute :skull:
+            assert (
```

Ruff (0.8.2)

Use of `assert` detected

(S101)